aspnetcore/fundamentals/url-rewriting.md

```diff
@@ -0,0 +1,295 @@
+* HTTP_ACCEPT
+* HTTP_CONNECTION
+* HTTP_COOKIE
+* HTTP_HOST
+* HTTP_REFERER
+* HTTP_URL
+* HTTP_USER_AGENT
+* HTTPS
+* LOCAL_ADDR
+* QUERY_STRING
+* REMOTE_ADDR
+* REMOTE_PORT
+* REQUEST_FILENAME
+* REQUEST_URI
+
+> [!NOTE]
+> You can also obtain an `IFileProvider` via a `PhysicalFileProvider`. This approach may provide greater flexibility for the location of your rewrite rules files. Make sure that your rewrite rules files are deployed to the server at the path you provide.
+> ```csharp
+> PhysicalFileProvider fileProvider = new PhysicalFileProvider(Directory.GetCurrentDirectory());
+> ```
+
+#### Method-based rule
+Use `Add(Action\<RewriteContext\> applyRule)` to implement your own rule logic in a method. The `RewriteContext` exposes the `HttpContext` for use in your method. The `context.Result` determines how additional pipeline processing is handled.
+
+`context.Result` | Action
+--- | ---
+`RuleResult.ContinueRules` (default) | Continue applying rules
+`RuleResult.End Response` | Stop applying rules and send the response
+`RuleResult.SkipRemainingRules` | Stop applying rules and send the context to the next middleware
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=6)]
+
+The sample application demonstrates a method that redirects requests for paths that end with **.xml**. If you make a request for **/file.xml**, it's redirected to **/xmlfiles/file.xml**. The status code is set to 301 (Moved Permanently). For a redirect, you should explicitly set the status code of the response; otherwise, a 200 (OK) status code will be returned and the redirect might not occur on the client.
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet2)]
+
+Original Request: **http://localhost:5000/file.xml**
+
+![Browser window with Developer Tools tracking the requests and responses for file.xml](url-rewriting/_static/add_redirect_xml_requests.png)
+
+#### `IRule`-based rule
+Use `Add(IRule)` to implement your own rule logic in a class that derives from `IRule`. Using an `IRule` provides greater flexibility over using the method-based rule approach. Your derived class may include a constructor, where you can pass in parameters for the `ApplyRule` method.
+
+[!code-csharp[Main](url-rewriting/sample/Startup.cs?name=snippet1&highlight=7-8)]
+
+The values of the parameters in the sample application for the `extension` and the `target` URL are checked to meet several conditions. The `extension` must contain a value, and the value must be **.png**, **.jpg**, or **.gif**. If the `target` URL isn't an absolute URL, an `ArgumentException` is thrown. Placing these values into fields, they're used in the rule logic. If you make a request for **/image.png**, it's redirected to **/png-images/image.png**. If you make a request for **/image.jpg**, it's redirected to **/jpg-images/image.jpg**. The status code is set to 301 (Moved Permanently), and the `context.Result` is set to stop processing rules and send the response.
```

Comment:
The description doesn't even require that sentence. Great catch btw --- We updated the sample but I didn't fix the description. That's fixed now.